core/src/main/scala/cats/data/HashMap.scala

```diff
@@ -267,7 +269,41 @@ object HashMap extends HashMapInstances {
   */
   final def fromSeq[K, V](seq: Seq[(K, V)])(implicit hashKey: Hash[K]): HashMap[K, V] = {
     val rootNode = seq.foldLeft(Node.empty[K, V]) { case (node, (k, v)) =>
-      node.add(k, improve(hashKey.hash(k)), v, 0)
+      node.updated(k, improve(hashKey.hash(k)), v, 0)
+    }
+    new HashMap(rootNode)
+  }
+
+  /**
+  * Creates a new [[cats.data.HashMap]] which contains all elements of `iterable`.
+  *
+  * @param iterable the iterable source of elements to add to the [[cats.data.HashMap]].
+  * @param hashKey the [[cats.kernel.Hash]] instance used for hashing values.
+  * @return a new [[cats.data.HashMap]] which contains all elements of `seq`.
+  */
+  final def fromIterableOnce[K, V](iterable: IterableOnce[(K, V)])(implicit hashKey: Hash[K]): HashMap[K, V] = {
+    iterable match {
+      case seq: Seq[(K, V) @unchecked] =>
+        fromSeq(seq)
+      case notSeq =>
+        val rootNode = notSeq.iterator.foldLeft(Node.empty[K, V]) { case (node, (k, v)) =>
+          node.updated(k, improve(hashKey.hash(k)), v, 0)
+        }
+        new HashMap(rootNode)
+    }
+  }
+
+  /**
+  * Creates a new [[cats.data.HashMap]] which contains all elements of `fkv`.
+  *
+  * @param fkv the [[cats.Foldable]] structure of elements to add to the [[cats.data.HashMap]].
+  * @param F the [[cats.Foldable]] instance used for folding the structure.
+  * @param hashKey the [[cats.kernel.Hash]] instance used for hashing values.
+  * @return a new [[cats.data.HashMap]] which contains all elements of `seq`.
+  */
+  final def fromFoldable[F[_], K, V](fkv: F[(K, V)])(implicit F: Foldable[F], hashKey: Hash[K]): HashMap[K, V] = {
```

Comment:
I guess we could weaken this constraint to `UnorderedFoldable`

Reply:
yeah, the order matters because you have to pick one of the values in the case of key collision.